src/actions/arcActions.ts

```diff
@@ -186,11 +191,15 @@ export async function getDAOData(avatarAddress: string, currentAccountAddress: s
   const executedProposals = await (await contributionRewardInstance.getExecutedProposals(daoInstance.avatar.address))({}, { fromBlock, toBlock }).get();
   const proposals = [...votableProposals, ...executedProposals];
 
-  // Get all proposals' details like title and description from the server
+  // Get all proposals' details like title and url from the server
   let serverProposals: { [key: string]: any } = {};
   try {
-    const results = await axios.get(process.env.API_URL + '/api/proposals?filter={"where":{"daoAvatarAddress":"' + avatarAddress + '"}}');
-    serverProposals = _.keyBy(results.data, "arcId");
+    const promises = proposals.map(async (proposal) => {
+      const ipfsHash = ipfs.hexToHash(proposal.contributionDescriptionHash)
```

Comment:
plz wait with that, as we have many different branches in parallel now and if we start liting on one branch we may run into trouble merging them

Reply:
also I'd vote for configurating the linter to *never* use semicolons :-)

Reply:
More important, though: why do we need this conversino "hextohash"?

Reply:
The IPFS file-id is a multihash encoded as base58 while the data stored on the blockchain is stored in hex. This function handles the conversion between raw hex in blockchain notion & the multihash required for interacting w ipfs nodes.

& If we want to keep the convention of adding unnecessary characters to the end of each line but don't want to use semicolons, how about we compromise & end each line with happy comments? `var isNiceCompromise = true || false // :)`

Reply:
Regarding linting, I would vote for being consistent across DAOstack projects.  Here is the one from arc.js which I think has been working nicely for a long time:

https://github.com/daostack/arc.js/blob/master/tslint.json

Reply:
@bohendo: What do you mean "the data stored on the blockchain is stored in hex"? It is just a string (isn't it? If not, it should be :-))

Reply:
@jellegerbrandy Ultimately, all data is stored as binary whether it's stored in Ethereum storage or being passed around as an IPFS hash

Popular ethereum tools like web3.js will, by default, interpret this chunk of binary data (stored on the blockchain as a bytes32 I believe) as a hex string.

Popular IPFS tools like the default HTTP interface, expect to receive a chunk of binary data in the form of an IPFS hash aka base58 encoded w a hash-specifying prefix.

Reply:
well yes of course it is binary in the end, my question was a retorical and not very well phrased, sorry about that.

My point is that there is no need to convert from "raw hex in blockchain" to "mutliahsh required for interacting with ipfs nodes". You can just use the string that IPFS gives you, and pass that without changing it to the web3 library (for writing), read the same string back with web3 and give that string, unchanged, to your ipfs interface and get the file. 

The fact that this string happends to be a base58-encoded multihash is just not relevant at this level of the code. (I know, I wrote very similar code myself about 12 months ago)

Reply:
Strings in solidity are UTF-8 encoded and the contract reserves 32 bytes for the descriptionHash.

An IPFS hash represented as UTF-8 consumes 46 bytes of data. This string *could* be saved to the blockchain as-is but not within 32 bytes of storage.

Or, if we base58 decode it and drop the multihash prefix, we get 32 bytes of data which fits cleanly in this storage slot.

Reply:
Ok, I see now where you are coming from. I had not realized that the contracts used as `bytes32` field for the ipfs hash.

I think this is a mistake and should be changed. There is no guarantee that IPFS hashes fit in 32 bytes (indeed, as you notice,  they do not). It is **not** Ok to drop the multihash prefix - this is meaningful information that encodes the hashing schema and length. IFS can and WILL  change the default hashing schema in the future: https://github.com/ipfs/ipfs/issues/337

I'll create an issue in the arc repo to change the type of the hash field, if you agree with my reasoning.

Reply:
https://github.com/daostack/arc/issues/577

Reply:
I'm glad you created this issue so we can make sure we're thinking about a better long-term solution. I definitely agree with your reasoning.

I'm not thrilled about dropping the prefix either but it seems like, given the current the contracts, this is a viable short term solution. Until the contracts change, I'm going to assume that this hacky solution works.

Important to note: we're already handling legacy data. When IPFS upgrades & breaks these old prefix-less hashes, we can incorporate them into the legacy data the same way we incorporated the data that was never saved to IPFS. So, this solution might not be future proof but it would be trivial to transition it to a future proof solution once one's available.

Reply:
So until we migrate to new contracts, we have no choice but to use this not-so-future-proof way of storing the IPFS hash right?